### PR TITLE
feat: integrating numscript rewrite

### PR DESCRIPTION
**Pull request by ascandone**
Changelog:
* use the interpreter in https://github.com/formancehq/numscript under a feature flag
* run e2e tests with both versions

The history is a bit messed up after the last rebase I did but should mostly make sense so you can read commit by commit

### CODE REVIEW

File: internal/controller/system/controller.go [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+	}
+	for _, opt := range append(defaultOptions, opts...) {
+		opt(ret)
+	}
+	return ret
+}
+
+type Option func(ctrl *DefaultController)
+
+func WithUpdateParser(updateParser func(oldParser ledgercontroller.NumscriptParser) ledgercontroller.NumscriptParser) Option {
```
Comment (gfyrag) [inline internal/controller/system/controller.go]: You don't need a so "complex" option. Just a `WithUpdateParser(ledgercontroller.NumscriptParser)` feels ok.
And decorating plumber in the module.go.

File: test/e2e/api_bulk.go [hunk 2]
```diff
@@ -3,188 +3,208 @@
...
+			})
+			BeforeEach(func() {
+				err := CreateLedger(ctx, testServer.GetValue(), operations.V2CreateLedgerRequest{
+					Ledger: "default",
+				})
+				Expect(err).To(BeNil())
+			})
+			When("creating a bulk on a ledger", func() {
+				var (
+					now   = time.Now().Round(time.Microsecond).UTC()
```
Comment (gfyrag) [inline test/e2e/api_bulk.go:55]: Not a big deal but there is wrapper around `time` package in go-libs, which already round dates at microseconds (postgres precision)

Comment (gfyrag) [reply test/e2e/api_bulk.go:55]: and set UTC too

**Review by gfyrag — APPROVED** [review]
Approved but i would be grateful if you can change the decorating stuff around the parser.
Keep the option simple, and make the plumber in the module.go file.

File: test/e2e/api_bulk.go [hunk 3]
```diff
@@ -3,187 +3,207 @@
...
-				components.CreateV2BulkElementCreateTransaction(components.V2BulkElementCreateTransaction{
-					Data: &components.V2PostTransaction{
-						Metadata: map[string]string{},
+	for _, data := range []struct {
+		description      string
+		numscriptRewrite bool
+	}{
+		{"default", false},
+		{"numscript rewrite", true},
+	} {
```
Comment (ascandone) [inline test/e2e/api_bulk.go:28]: Here the only diff is that I wrapped everything in this for loop

File: test/e2e/api_bulk.go [hunk 4]
```diff
@@ -3,187 +3,207 @@
...
+			)
+
+			testServer := NewTestServer(func() Configuration {
+				return Configuration{
+					PostgresConfiguration:        db.GetValue().ConnectionOptions(),
+					Output:                       GinkgoWriter,
+					Debug:                        debug,
+					NatsURL:                      natsServer.GetValue().ClientURL(),
+					BulkMaxSize:                  bulkMaxSize,
+					ExperimentalNumscriptRewrite: data.numscriptRewrite,
```
Comment (ascandone) [inline test/e2e/api_bulk.go:44]: Not that here it's using the feature flag value

File: test/e2e/api_transactions_create.go [hunk 5]
```diff
@@ -22,466 +23,504 @@ import (
...
-		})
-		Expect(err).To(BeNil())
-	})
+	for _, data := range []struct {
+		description      string
+		numscriptRewrite bool
+	}{
+		{"default", false},
+		{"numscript rewrite", true},
+	} {
```
Comment (ascandone) [inline test/e2e/api_transactions_create.go:32]: Same as before, here the only diff is wrapping all in a loop

**Pull request merged**